packages/client/src/pages/customer_area/views/Book.tsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import { IntervalCardGroup, SlotsDayContainer } from "@eisbuk/ui";
+import { SlotInterface } from "@eisbuk/shared";
+
+import BookingsCountdownContainer from "@/controllers/BookingsCountdown";
+
+import { getSlotsForBooking } from "@/store/selectors/bookings";
+
+import { bookInterval } from "@/store/actions/bookingOperations";
+import { openModal } from "@/features/modal/actions";
+
+import { getSecretKey } from "@/utils/localStorage";
+
+const BookView: React.FC = () => {
+  const daysToRender = useSelector(getSlotsForBooking);
+
+  const { handleBooking, handleCancellation } = useBooking();
+
+  return (
+    <>
+      <BookingsCountdownContainer />
+
+      {daysToRender.map(({ date, slots }) => (
+        <SlotsDayContainer key={date} date={date}>
+          {slots.map(({ interval, ...slot }) => (
```

Comment:
Should slots first be sorted by their first intervals start time?
The intervals within each `IntervalCardGroup` are ordered by duration, but these groups are often not in chronological order - see screens below.

<img width="732" alt="Screen Shot 2022-07-05 at 12 05 11 PM" src="https://user-images.githubusercontent.com/62590503/177305633-e047e716-6d2d-45b5-ab79-57d6d897e10d.png">
<img width="959" alt="Screen Shot 2022-07-05 at 12 08 11 PM" src="https://user-images.githubusercontent.com/62590503/177305659-8554ed9e-c477-4143-a801-0ff07a4c0a73.png">

Reply:
I share your "frustration" with the way these things are...and am also confused as to how to go about it...should we just remove the `BookingCardGroup`* and have the intervals ordered in a way that the longest are first (regarding of which slot they belong to) or order the slots in ascending order and then apply the ordering within (in this case I'd keep the `IntervalCard` group). I've already noted this for enhancement and would like this to go in ASAP so that Fadwa can start on e2e while someone else (me) works on enhancements/fixes.

*\*it isn't all too necessary without the faded logic and serves only as a minor abstraction to keep things cleaner*